pkg/kubelet/util/pluginwatcher/README

```diff
@@ -0,0 +1,29 @@
+This folder contains a utility, pluginwatcher, for Kubelet to register
+different types of node-level plugins such as device plugins or CSI plugins.
+It discovers plugins by monitoring inotify events under the directory returned by
+kubelet.getPluginsDir(). Lets refer this directory as PluginsSockDir.
+For any discovered plugin, pluginwatcher issues Registration.GetInfo grpc call
+to get plugin type, name and supported service API versions. For any registered plugin type,
+pluginwatcher calls the registered callback function with the received plugin
+name, supported service API versions, and the full socket path. The Kubelet
+component that receives this callback can acknowledge or reject the plugin
+according to its own logic, and use the socket path to establish its service
+communication with any API version supported by the plugin.
+
+Here are the general rules that Kubelet plugin developers should follow:
+- Run as priviledged. Currently creating socket under PluginsSockDir requires
+  priviledged access.
```

Comment:
I don't think that's true? Doesn't that just require running as root & a HostPath mount? I think it's likely that plugins will still need to be privileged, but for different reasons...

Reply:
+1 If all that's needed is access to a root owned directory, that could come in the form of running as gid `0` too right?

Reply:
Depending on the access permissions, but I'd recommend requiring a root uid (i.e. `700` permissions). The k8s controls limit access to gid 0 are still alpha.